pkg/dynamicinstrumentation/diconfig/mem_config_manager.go

```diff
@@ -26,7 +26,7 @@ import (
 type ReaderConfigManager struct {
 	sync.Mutex
```

Comment:
This mutex being exported from this struct is confusing to me: does it apply to the public members? Why else would it be exported? Anyways, that's more of a drive-by nit, and if this code is to stick around for very long we'll want to clean up this sort of thing...

Reply:
That's unrelated to this PR per-se, but I agree we may refactor around before we introduce the revamp of Go DI